src/FieldElementSubsystems/Connections/EulynxBaseline4R1/PointToInterlockingConnection.cs

```diff
@@ -14,59 +14,77 @@ public class PointToInterlockingConnection : IPointToInterlockingConnection
-        CancellationToken stoppingToken,
-        int timeoutDuration = 10000)
+        CancellationToken stoppingToken)
     {
-        _timeoutDuration = timeoutDuration;
         _stoppingToken = stoppingToken;
-        _timeout = new CancellationTokenSource();
         _logger = logger;
         CurrentConnection = null;
 
         var config = configuration.GetSection("PointSettings").Get<PointConfiguration>() ?? throw new Exception("No configuration provided");
         _localId = config.LocalId;
         _remoteId = config.RemoteId;
+        _pdiVersion = config.PDIVersion;
+        _checksum = config.PDIChecksum.HexToByteArray();
+        if (_checksum.Length != 2) throw new InvalidOperationException($"Invalid checksum length. Expected 2 bytes was ${_checksum.Length}.");
         Configuration = config;
         _overrideMessages = Channel.CreateUnbounded<byte[]>();
     }
 
-    public void Connect(IConnection connection)
+    public IPointToInterlockingConnection Connect(IConnection connection)
     {
-        ResetTimeout();
         CurrentConnection = connection;
+        return this;
     }
 
-    public async Task<bool> InitializeConnection(GenericPointState state, bool observeAbilityToMove, CancellationToken cancellationToken)
+    public async Task<bool> InitializeConnection(GenericPointState state, bool observeAbilityToMove, bool simulateTimeout, CancellationToken cancellationToken)
     {
-        if (await ReceiveMessage<PointPdiVersionCheckCommand>(cancellationToken) == null)
+        var versionCheckReceived = await ReceiveMessage<PointPdiVersionCheckCommand>(cancellationToken);
+        
+        if (versionCheckReceived == null)
         {
             _logger.LogError("Unexpected message.");
             return false;
         }
 
-        var versionCheckResponse = new PointPdiVersionCheckMessage(_localId, _remoteId, PointPdiVersionCheckMessageResultPdiVersionCheck.PDIVersionsFromReceiverAndSenderDoMatch, /* TODO */ 0, 0, Array.Empty<byte>());
+        if (!CheckPDIVersionReceived(versionCheckReceived.PdiVersionOfSender))
+        {
+            // Eu.Gen-SCI.445
+            // Eu.SCI-XX.PDI.91 - Eu.SCI-XX.PDI.94
+            _logger.LogError("Version check failed.");
+            var versionCheckFailedResponse = new PointPdiVersionCheckMessage(_localId, _remoteId, PointPdiVersionCheckMessageResultPdiVersionCheck.PDIVersionsFromReceiverAndSenderDoNotMatch, _pdiVersion, 0, []);
+            await SendMessage(versionCheckFailedResponse);
+            return false;
+        }
+
+        var versionCheckResponse = new PointPdiVersionCheckMessage(_localId, _remoteId, PointPdiVersionCheckMessageResultPdiVersionCheck.PDIVersionsFromReceiverAndSenderDoMatch, _pdiVersion, (byte)_checksum.Length, _checksum);
+        
         await SendMessage(versionCheckResponse);
 
+        if (simulateTimeout)
+        {
+            // Never send the missing initialization messages
+            return false;

```

Comment:
Doesn't this abort the RaSTA connection? That wouldn't be the intended behavior, right?